core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java

```diff
@@ -192,12 +193,24 @@ void logResponseFirstBytesTransferred() {
 
         @Override
         public void run() {
-            final ResponseTimeoutException cause = ResponseTimeoutException.get();
-            delegate.close(cause);
-            logBuilder.endResponse(cause);
+            final Runnable responseTimeoutHandler;
+            if (logBuilder != RequestLogBuilder.NOOP) {
```

Comment:
After thinking a little bit, I thought `!(logBuilder instanceof DefaultRequestLog)` might be a little bit better? :sweat_smile: